File: 125-ValidPalindrome/125-ValidPalindrome.java
// Last updated: 26/03/2026, 16:22:19
class Solution {
    public boolean isPalindrome(String s) {
        s=s.replaceAll("[^a-zA-Z0-9\\s]", "");
        s=s.toLowerCase();
        char[] ch=s.toCharArray();
        for(int i=0,j=ch.length-1;i<ch.length/2;i++,j--)
        {   
            while((ch[i]+"").equals(" "))
            i++;
            while((ch[j]+"").equals(" "))
            j--;
            if(ch[i]==ch[j])
                continue;
            return false;
        }
        return true;
        
    }
}

File: 125-ValidPalindrome/125-ValidPalindrome.py
# Last updated: 16/06/2026, 18:22:44
1class Solution(object):
2    def singleNumber(self, nums):
3        nums.sort()
4        for i in range(0,len(nums)-1,2):
5            if nums[i]!=nums[i+1]:
6                return nums[i]
7        return nums[len(nums)-1]
8
9        